### PR TITLE
feat(platform): support socks5 proxy

### PR DESCRIPTION
**Pull request by leoryu**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespace from that line:
>
> /kind api-change
> /kind bug
> /kind cleanup
> /kind deprecation
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

File: pkg/util/ssh/socks5_proxy.go [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+	"tkestack.io/tke/pkg/util/log"
+)
+
+type SOCKS5 struct {
+	Host        string
+	Port        int
+	DialTimeOut time.Duration
+}
+
+func (sk SOCKS5) ProxyConn(targetAddr string) (net.Conn, func(), error) {
```
Comment (wl-chen) [inline pkg/util/ssh/socks5_proxy.go:36]: sk SOCKS5 -> sk *SOCKS5 ?

File: pkg/util/ssh/socks5_proxy.go [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+		ch <- r
+	}()
+	select {
+	case r := <-ch:
+		return r.conn,
+			func() {
+				r.conn.Close()
+			},
+			r.err
+	case <-time.After(sk.DialTimeOut):
```
Comment (wl-chen) [inline pkg/util/ssh/socks5_proxy.go:66]: 如果是超时，可不可以先调一下 r.conn.Close()，再返回

File: pkg/util/ssh/socks5_proxy.go [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+	"time"
+
+	"golang.org/x/net/proxy"
+	"tkestack.io/tke/pkg/util/log"
+)
+
+type SOCKS5 struct {
+	Host        string
+	Port        int
+	DialTimeOut time.Duration
```
Comment (wl-chen) [inline pkg/util/ssh/socks5_proxy.go:33]: 我看实际调的 proxy.SOCKS5 接口可以传 auth值，里面是 username 和 password，虽然是选填，但是这里先加上方便后面扩展？

File: pkg/util/ssh/socks5_proxy.go (hunk 1 above)
Comment (leoryu) [reply pkg/util/ssh/socks5_proxy.go:36]: We don't need to change sk's content, we just read it's content, so no pointer is better.

File: pkg/util/ssh/socks5_proxy.go (hunk 2 above)
Comment (leoryu) [reply pkg/util/ssh/socks5_proxy.go:66]: aggree

File: pkg/util/ssh/socks5_proxy.go (hunk 3 above)
Comment (leoryu) [reply pkg/util/ssh/socks5_proxy.go:33]: ok

Comment (leoryu) [reply pkg/util/ssh/socks5_proxy.go:35]: ok

File: pkg/util/ssh/socks5_proxy.go (hunk 1 above)
Comment (leoryu) [reply pkg/util/ssh/socks5_proxy.go:38]: since we just read the sk's content, not write, to prevent someone changing its content, no pointer is better.

**Pull request merged**